dookug-common/dookug-common-pdfsignature/src/main/java/hu/icellmobilsoft/common/openpdfsign/PdfSigner.java

```diff
@@ -0,0 +1,374 @@
+     * @throws BaseException
+     *             on error
+     */
+    public void signPdf(Path originalPdfFile, SignatureProfileDto profile, OutputStream outputStream) throws BaseException {
+        DSSDocument signedDocument = signPdf(originalPdfFile, profile);
+        log.debug(DOCUMENT_SIGNING_COMPLETE);
+        try {
+            signedDocument.writeTo(outputStream);
+        } catch (IOException e) {
+            throw new TechnicalException(CoffeeFaultType.OPERATION_FAILED, e.getLocalizedMessage(), e);
+        } finally {
+            boolean result;
+            try {
+                result = Files.deleteIfExists(originalPdfFile);
+                if (!result) {
+                    log.warn("original PDF file [{0}] couldnt be removed!", originalPdfFile);
+                }
+            } catch (IOException e) {
+                log.warn("original PDF file [{0}] couldnt be removed! [{1}]", originalPdfFile, e.getLocalizedMessage());
+            }
+        }
+    }
+
+    /**
+     * Sign the pdf and return as stream
+     * 
+     * @param pdfFile
+     *            the pdf file to sign
+     * @param keyStore
+     *            the keystore with the signer key (ideally with keyAlias: {@value KeystoreLoader#DEFAULT_ALIAS})
+     * @param keyStorePassword
+     *            the password of the keystore file
+     * @param profile
+     *            the configuration to override the server default configuration
+     * @param binaryOutput
+     *            the signed PDF document as an {@link OutputStream}
+     * @throws BaseException
+     *             on error
+     */
+    public void signPdf(Path pdfFile, byte[] keyStore, char[] keyStorePassword, SignatureProfileDto profile, OutputStream binaryOutput)
+            throws BaseException {
+        DSSDocument signedDocument = signPdf(pdfFile, profile);
+        log.debug(DOCUMENT_SIGNING_COMPLETE);
+        try {
+            signedDocument.writeTo(binaryOutput);
+        } catch (IOException e) {
+            throw new TechnicalException(CoffeeFaultType.OPERATION_FAILED, e.getLocalizedMessage(), e);
+        }
+    }
+
+    private DSSDocument signPdf(Path pdfFile, SignatureProfileDto profile) throws BaseException {
+
+        boolean visibleSignature = profile.getDssPage() != 0;
+
+        // load PDF file in DSSDocument format
+        DSSDocument toSignDocument = new PathDocument(pdfFile);
+
+        // load certificate and private key
+        byte[] keystore = keystoreLoader
+                .loadConfiguredFile(profile.getKeystoreType(), profile.getKeystore(), profile.getKeystorePassword().toCharArray());
+        try (JKSSignatureToken signingToken = new JKSSignatureToken(
+                keystore,
+                // new KeyStore.PasswordProtection(keystoreLoader.getCurrentKeystorePassword()))) {
+                new KeyStore.PasswordProtection(profile.getKeystorePassword().toCharArray()))) {
+            log.debug("Keystore created for signing");
```

Comment:
ezekbe a debug üzenetekbe hasznos lenne pár info, pl melyik pdfFile-hoz, vagy milyen profillal. Ez a lentebbi log üzenetekre is érvényes.

Reply:
beletettem